Controllers/GetAllPayment.controller.js

```diff
@@ -0,0 +1,34 @@
+const Payment = require('../Models/Payment.model');
+const Booking = require('../Models/bookingData.model');
+const UserModel = require("../Models/user.model")
+
+exports.findAll = async(req, res) => {
+
+    const user = await UserModel.findOne({ userId: req.userId }).exec();
+    if (!user) {
+        return res.status(404).send({
+            message: "User Not Found"
+        })
+    }
+    let queryObj = {};
+    if (user.role != "ADMIN") {
+        const booking = await Booking.find({ user }).exec();
+
+        if (!booking) {
+            return res.status(404).send({
+                message: 'Bookings Not Found'
+            })
+        }
+
+        queryObj._id = booking._id;
```

Comment:
booking response object is an array. Use a map to create an array of booking ids.